probe/endpoint/ebpf.go

```diff
@@ -287,10 +314,45 @@ func (t *EbpfTracker) isDead() bool {
 
 func (t *EbpfTracker) stop() {
 	t.Lock()
-	alreadyDead := t.dead
-	t.dead = true
+	alreadyDead := t.dead || t.stopping
+	t.stopping = true
 	t.Unlock()
-	if !alreadyDead && t.tracer != nil {
-		t.tracer.Stop()
+
+	// Do not call tracer.Stop() in this thread, otherwise tracer.Stop() will
+	// deadlock waiting for this thread to pick up the next event.
```

Comment:
The deadlock was not with tcptracer-bpf's channel `t.stopChan` but with gobpf's channels:
- `t.perfMapIPV4.PollStop()` / `pm.pollStop <- true` ([perf.go#L244](https://github.com/iovisor/gobpf/blob/51b4a5bc15ab504f9efb7702591ae04bc6c6f068/elf/perf.go#L244))
- `pm.receiverChan <- incoming.bytesArray[0]` ([perf.go#L229](https://github.com/iovisor/gobpf/blob/51b4a5bc15ab504f9efb7702591ae04bc6c6f068/elf/perf.go#L229))

I could reliably reproduce the deadlock before using pprof to print the go routine stacks (following this [doc](https://stackoverflow.com/questions/19094099/how-to-dump-goroutine-stacktraces/19145992#19145992)).

Maybe this should be fixed at the gobpf level but I don't immediately see how, because the caller of gobpf's `PollStop()` needs to know when it's done...